insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -1261,150 +1278,240 @@ private void processDateTimeChangedEvent(DateTimeChangedEvent event) {
+        long duration;
         switch (event.getNewValue()) {
             case STARTED:
-                pumpID.eventType = "PumpStarted";
-                pumpStartedEvents.add(pumpID);
-                if (sp.getBoolean("insight_log_operating_mode_changes", false))
+                lastStartEvent = timestamp + 10000L;                        // I don't now the reason of 10s offset, so I keep it as it was in original Insight Driver
+                if (sp.getBoolean(R.string.key_insight_log_operating_mode_changes, false))
                     logNote(timestamp, resourceHelper.gs(R.string.pump_started));
+                duration = lastStartEvent - lastStopEvent;
+                if (duration > T.mins(1).msecs() && duration < T.hours(24).msecs())
+                    pumpSync.syncTemporaryBasalWithPumpId(
+                            lastStopEvent,
+                            0.0,
+                            duration,
+                            false,
+                            PumpSync.TemporaryBasalType.NORMAL,
+                            lastStopEvent,
+                            PumpType.ACCU_CHEK_INSIGHT,
+                            serial);
+                lastStartEvent = 0;
+                aapsLogger.debug(LTag.PUMP, "XXXX event START Event TimeStamp: " + lastStartEvent + " HMS: " + dateUtil.dateAndTimeAndSecondsString(lastStartEvent));
                 break;
             case STOPPED:
-                pumpID.eventType = "PumpStopped";
-                if (sp.getBoolean("insight_log_operating_mode_changes", false))
+                lastStopEvent = Math.abs(lastStopEvent - timestamp) < 10000 ? lastStopEvent : timestamp;                                            // timestamp could vary between 2 readhistory, always keep first value
+                duration = dateUtil.now() + T.mins(15).msecs() - lastStopEvent;                                                                     // record 15 min zero temp only if duration is more than 1 min
+                if (duration > T.mins(1).msecs())                                                                                                   // set a zero temp if pump is stopped more than 1 min
+                    pumpSync.syncTemporaryBasalWithPumpId(
+                            lastStopEvent,
+                            0.0,
+                            duration,
+                            false,
+                            PumpSync.TemporaryBasalType.NORMAL,
+                            lastStopEvent,
+                            PumpType.ACCU_CHEK_INSIGHT,
+                            serial
+                    );
+                lastStartEvent = 0;
+                aapsLogger.debug(LTag.PUMP, "XXXX event STOP: " + lastStopEvent + " HMS: " + dateUtil.dateAndTimeAndSecondsString(lastStopEvent) + " ZeroTemp Duration (min): " + (duration)/60000);
+                if (sp.getBoolean(R.string.key_insight_log_operating_mode_changes, false))
```

Comment:
Why do you save a TBR here? You can only determine the duration once the pump is started again by searching for the previous stop event. Also, you should not save a fake TBR if the previous operating mode change was a pause event. If the pump is started again after having been in pause mode (for a maximum of 15 minutes), all the missing insulin from that time frame is delivered.

Reply:
My idea here was to immediatly start zero TBR only on Stop event (not for pause event), even if Start event is not in History (duration now + 15 min so duration updated on each readHistory).
=> removed on #502